cpp/src/io/parquet/reader_impl_chunking.cu

```diff
@@ -359,6 +359,28 @@ void reader::impl::create_global_chunk_info()
   auto const num_input_columns = _input_columns.size();
   auto const num_chunks        = row_groups_info.size() * num_input_columns;
 
+  // Mapping of input column to page index column
+  std::vector<size_type> column_mapping;
+
+  if (_has_page_index and not row_groups_info.empty()) {
+    // use first row group to define mappings (assumes same schema for each file)
+    auto const& rg      = row_groups_info[0];
+    auto const& columns = _metadata->get_row_group(rg.index, rg.source_index).columns;
+    column_mapping.resize(num_input_columns);
+    std::transform(
+      _input_columns.begin(), _input_columns.end(), column_mapping.begin(), [&](auto const& col) {
+        // translate schema_idx into something we can use for the page indexes
+        if (auto it = std::find_if(
+              columns.begin(),
+              columns.end(),
+              [&col](auto const& col_chunk) { return col_chunk.schema_idx == col.schema_idx; });
```

Comment:
are columns maybe sorted by schema_idx?

Reply:
The input columns can be in any order depending on user input. The `schema_idx` is determined by the file schema IIRC.

Reply:
In fact this is similar to what `_metadata->get_column_metadata(rg.index, rg.source_index, col.schema_idx)` is doing below.

Reply:
Trying to find a(n easy) way to avoid the quadratic complexity here. Should be fine as-is.